var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -174,32 +174,25 @@ def cmake_args(self):
             # However, with spack it is preferrable to have a proper external
             # dependency, so we disable it.
             self.define("DD4HEP_LOAD_ASSIMP", False),
-            "-DCMAKE_CXX_STANDARD={0}".format(cxxstd),
-            "-DBUILD_TESTING={0}".format(self.run_tests),
-            "-DBOOST_ROOT={0}".format(spec["boost"].prefix),
-            "-DBoost_NO_BOOST_CMAKE=ON",
+            self.define("CMAKE_CXX_STANDARD", cxxstd),
+            self.define("BUILD_TESTING", self.run_tests),
+            self.define("BOOST_ROOT", spec["boost"].prefix),
+            self.define("Boost_NO_BOOST_CMAKE", True),
         ]
-        subpackages = []
-        if spec.satisfies("+ddg4"):
-            subpackages += ["DDG4"]
-        if spec.satisfies("+ddcond"):
-            subpackages += ["DDCond"]
-        if spec.satisfies("+ddcad"):
-            subpackages += ["DDCAD"]
-        if spec.satisfies("+ddrec"):
-            subpackages += ["DDRec"]
-        if spec.satisfies("+dddetectors"):
-            subpackages += ["DDDetectors"]
-        if spec.satisfies("+ddalign"):
-            subpackages += ["DDAlign"]
-        if spec.satisfies("+dddigi"):
-            subpackages += ["DDDigi"]
-        if spec.satisfies("+ddeve"):
-            subpackages += ["DDEve"]
-        if spec.satisfies("+utilityapps"):
-            subpackages += ["UtilityApps"]
-        subpackages = " ".join(subpackages)
-        args += [self.define("DD4HEP_BUILD_PACKAGES", subpackages)]
+
+        packages = [
+            "DDG4",
+            "DDCond",
+            "DDCAD",
+            "DDRec",
+            "DDDetectors",
+            "DDAlign",
+            "DDDigi",
+            "DDEve",
+            "UtilityApps",
+        ]
+        enabled_packages = [p for p in packages if self.spec.variants[package.lower()].value]
```

Comment:
This breaks the DD4hep package installation, because `package` is not a variable, it needs to read

```python
enabled_packages = [p for p in packages if self.spec.variants[p.lower()].value]
```

Am I the first one to test this?

Reply:
Oops, my bad!

Reply:
This is why we need the hep cloud pipelines, since they would have caught this before the merge: https://gitlab.spack.io/spack/spack/-/jobs/14330604 :-)

Please give #40932 a review and thumbs up :-)

We can tweak what's included there (incl adding `acts +traccc` and a cuda stack), but I want to get at least something merge soon.